src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,111 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleBatchBuffer;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Operator which splits a string-valued column on a Java regular expression and duplicates the input row with each
+ * segment of the split result.
+ * 
+ * E.g., (1, 2, "foo:bar:baz") -> (1, 2, "foo"), (1, 2, "bar"), (1, 2, "baz")
+ */
+public final class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Index of (string-valued) column to split on regex.
+   */
+  private final int splitColumnIndex;
+
+  /**
+   * Compiled regex to split input tuples.
+   */
+  private final Pattern pattern;
+
+  /**
+   * Buffer to hold finished and in-progress TupleBatches.
+   */
+  private TupleBatchBuffer outputBuffer;
+
+  /**
+   * 
+   * @param child child operator that data is fetched from
+   * @param splitColumnIndex index of string column to split using {@link #regex}
+   * @param regex regular expression to split value of column at {@link #splitColumnIndex}
+   */
+  public Split(@Nonnull final Operator child, final int splitColumnIndex, @Nonnull final String regex) {
+    super(child);
+    Preconditions.checkNotNull(child, "Child operator cannot be null");
```

Comment:
for these, just put the name of the `null` parameter – the message will be `NullPointerException: child` and the extra text is just a waste of your time and bytes :)